WordPress/openverse-frontend
# Replace focus-trap-vue with Vueuse implementation

## Fixes


Fixes #1752 by @ramadanomar

## Description


This PR copies VueUse integrations/useFocusTrap/component implementation to use the latest version of the upstream dependency, `focus-trap`.

It is clear that `focus-trap-vue` is not going to update the Vue-2 branch to add the fix for the linked issue. I wanted to use the `useFocusTrap` component from VueUse at first, but I couldn't solve the [Nuxt 2 import issue](https://github.com/vueuse/vueuse/issues/2140), so I vendored in the UseFocusTrap component. VueUse library is using MIT license, so it should be OK. Hopefully, after we move on to Vue 3, we can use the VueUse library instead of vendoring in the implementation.

## Testing Instructions

Run the app with `pnpm dev`, and on mobile screen, open the menu and filters modals. There should be no errors/warnings regarding the initial focus node in the console, and the focus should be handled correctly. When opening the modal, the first tabbable element should be focused. When you close the modal, you should be able to tab away from the modal trigger.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
